src/pie_utils/processor/document/split_into_partition.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import json
+import logging
+import re
+from typing import Any, Callable, Iterable, Iterator, Match
+
+from pytorch_ie.annotations import LabeledSpan
+
+from pie_utils.statistics import WithStatistics
+
+from ..document import DocumentWithPartition
+
+logger = logging.getLogger(__name__)
+
+
+def get_partitions_with_matcher(
+    document: DocumentWithPartition,
+    matcher: Callable[[str], Iterable[Match]],
+    label_group_id: int | None = None,  # = 1,
+    label_whitelist: list[str] | None = None,
+    skip_initial_partition: bool = False,  # = True
+    initial_partition_label: str = "initial part",
+) -> Iterator[LabeledSpan]:
+    """Spans are created starting with the beginning of matching entries end ending with the start
+    of the next matching one or the end of the document.
+
+    Entries with xml node names that are not in label_whitelist, if it is set, will not be
+    considered as match, i.e. their content will be added to the previous matching entry. If
+    label_group_id is set, the content of the respective match group will be taken as label.
+    Otherwise, it is set to None. If the flag skip_initial_partition is enabled, the content before
+    the first match is not added as a partition. Note that the initial partition will get None as
+    label since no matched element is available.
+    """
+    previous_start = previous_label = None
+    if not skip_initial_partition:
+        previous_start = 0
+        previous_label = initial_partition_label  # This is added here because if we want to keep initial partition then without
+        # setting a label here, None is added as label which result in exception. We can have a parameter for
+        # initial_label.
+    for match in matcher(document.text):
+        if label_group_id is not None:
+            start = match.start(label_group_id)
+            end = match.end(label_group_id)
+            label = document.text[start:end]
+        else:
+            label = None
+        if label_whitelist is None or label in label_whitelist:
+            if previous_start is not None:
+                end = match.start()
+                span = LabeledSpan(start=previous_start, end=end, label=previous_label)
+                yield span
+
+            previous_start = match.start()
+            previous_label = label
+
+    if previous_start is not None:
+        end = len(document.text)
+        span = LabeledSpan(start=previous_start, end=end, label=previous_label)
+        yield span
+
+
+class SplitDocumentToPartitions(WithStatistics):
```

Comment:
Let's rename that to `RegexPartitioner` and the files (test_)regex_partitioner.py.

Reply:
renamed in [906106e](https://github.com/ArneBinder/pie-utils/pull/21/commits/906106e0237869813aa0d3fe3070b05a231b804a)